drivers/media/i2c/mira220.c

```diff
@@ -0,0 +1,2087 @@
+		goto err_rpm_put;
+	}
+
+	__v4l2_ctrl_grab(mira220->vflip, true);
+	__v4l2_ctrl_grab(mira220->hflip, true);
+
+	return 0;
+
+err_rpm_put:
+	pm_runtime_put(&client->dev);
+	return ret;
+}
+
+static void mira220_stop_streaming(struct mira220 *mira220)
+{
+	struct i2c_client *client = v4l2_get_subdevdata(&mira220->sd);
+	int ret = 0;
+
+	/* Unlock controls for vflip and hflip */
+	__v4l2_ctrl_grab(mira220->vflip, false);
+	__v4l2_ctrl_grab(mira220->hflip, false);
+
+	ret = mira220_write_stop_streaming_regs(mira220);
+	if (ret) {
+		dev_err(&client->dev,
+			"Could not write the stream-off sequence");
+	}
+
+	pm_runtime_put(&client->dev);
+}
+
+static int mira220_set_stream(struct v4l2_subdev *sd, int enable)
+{
+	struct mira220 *mira220 = to_mira220(sd);
+	int ret = 0;
+
+	mutex_lock(&mira220->mutex);
+
+	if (enable) {
+		/*
+		 * Apply default & customized values
+		 * and then start streaming.
+		 */
+		ret = mira220_start_streaming(mira220);
+		if (ret)
+			goto err_unlock;
+	} else {
+		mira220_stop_streaming(mira220);
+	}
+
+	mutex_unlock(&mira220->mutex);
+
+	return ret;
+
+err_unlock:
+	mutex_unlock(&mira220->mutex);
+
+	return ret;
+}
+
+static int mira220_get_regulators(struct mira220 *mira220)
+{
+	struct i2c_client *client = v4l2_get_subdevdata(&mira220->sd);
+	unsigned int i;
+
+	for (i = 0; i < MIRA220_NUM_SUPPLIES; i++)
+		mira220->supplies[i].supply = mira220_supply_name[i];
+
+	return devm_regulator_bulk_get(&client->dev, MIRA220_NUM_SUPPLIES,
+				       mira220->supplies);
+}
+
+/* OTP power on */
+static int mira220_otp_power_on(struct mira220 *mira220)
+{
+	int ret;
+
+	ret = cci_write(mira220->regmap, MIRA220_OTP_CMD_REG,
+			MIRA220_OTP_CMD_UP, NULL);
+
+	return 0;
+}
+
+/* OTP power off */
+static int mira220_otp_power_off(struct mira220 *mira220)
+{
+	int ret;
+
+	ret = cci_write(mira220->regmap, MIRA220_OTP_CMD_REG,
+			MIRA220_OTP_CMD_DOWN, NULL);
+
+	return 0;
+}
+
+/* OTP power on */
+static int mira220_otp_read(struct mira220 *mira220, u8 addr, u8 offset,
+			    u8 *val)
+{
+	int ret;
+	u64 readback;
+
+	ret = cci_write(mira220->regmap, CCI_REG8(0x0086), addr, NULL);
+	ret = cci_write(mira220->regmap, CCI_REG8(0x0080), 0x02, NULL);
+	ret = cci_read(mira220->regmap, CCI_REG8(0x0082 + offset), &readback,
+		       NULL);
+	*val = readback & 0xFF;
+
+	return 0;
+}
+
+/* Verify chip ID */
+static int mira220_identify_module(struct mira220 *mira220)
+{
+	struct i2c_client *client = v4l2_get_subdevdata(&mira220->sd);
+	int ret;
+	u8 val;
+
+	mira220_otp_power_on(mira220);
+
+	usleep_range(100, 110);
+
+	ret = mira220_otp_read(mira220, 0x0d, 0, &val);
+	dev_err(&client->dev, "Read OTP add 0x0d with val %x\n", val);
+	ret = mira220_otp_read(mira220, 0x19, 0, &val);
+	dev_err(&client->dev, "Read OTP add 0x19 with val %x\n", val);
+	ret = mira220_otp_read(mira220, 0x19, 1, &val);
+	dev_err(&client->dev, "Read OTP add 0x19+1 with val %x\n", val);
+
+	mira220_otp_power_off(mira220);
+
+	return 0;
```

Comment:
If we never fail the identity check, then make it a void function.

You never check any of the returned values from `mira220_otp_read` which would be a basic identity check.